tests/resource/test_resource_setters.py

```diff
@@ -18,10 +18,258 @@ class TestResourceSetters(unittest.TestCase):
+        )
+        self.assertEqual(
+            self.resource_admin.alter_audit_by_access_level(
+                "TESTING",
+                "ELIJTEST",
+                alter="success",
+                control="failure",
+                update="all",
+                read="all",
+            ),
+            TestResourceConstants.TEST_RESOURCE_ALTER_AUDIT_BY_ACCESS_LEVEL_ALL_REQUEST_XML,
+        )
+
+    @patch("pyracf.resource.resource_admin.ResourceAdmin.extract")
+    def test_resource_admin_build_alter_audit_by_attempt_request(
+        self,
+        resource_admin_extract_mock: Mock,
+    ):
+        resource_admin_extract_mock.return_value = (
+            TestResourceConstants.TEST_EXTRACT_RESOURCE_GET_AUDIT_RULES
+        )
+        self.assertEqual(
+            self.resource_admin.alter_audit_by_attempt(
+                "TESTING", "ELIJTEST", failure="control"
+            ),
+            TestResourceConstants.TEST_RESOURCE_ALTER_AUDIT_BY_ATTEMPT_REQUEST_XML,
+        )
+
+    @patch("pyracf.resource.resource_admin.ResourceAdmin.extract")
+    def test_resource_admin_build_alter_audit_by_attempt_none_request(
+        self,
+        resource_admin_extract_mock: Mock,
+    ):
+        resource_admin_extract_mock.return_value = (
+            TestResourceConstants.TEST_EXTRACT_RESOURCE_GET_AUDIT_RULES
+        )
+        self.assertEqual(
+            self.resource_admin.alter_audit_by_attempt("TESTING", "ELIJTEST"),
+            TestResourceConstants.TEST_RESOURCE_ALTER_AUDIT_BY_ATTEMPT_NONE_REQUEST_XML,
+        )
+
+    @patch("pyracf.resource.resource_admin.ResourceAdmin.extract")
+    def test_resource_admin_build_alter_audit_by_attempt_multiple_request(
+        self,
+        resource_admin_extract_mock: Mock,
+    ):
+        resource_admin_extract_mock.return_value = (
+            TestResourceConstants.TEST_EXTRACT_RESOURCE_GET_AUDIT_RULES
+        )
+        self.assertEqual(
+            self.resource_admin.alter_audit_by_attempt(
+                "TESTING", "ELIJTEST", failure="control", all="read"
+            ),
+            TestResourceConstants.TEST_RESOURCE_ALTER_AUDIT_BY_ATTEMPT_MULT_REQUEST_XML,
+        )
+
+    @patch("pyracf.resource.resource_admin.ResourceAdmin.extract")
+    def test_resource_admin_build_alter_audit_by_attempt_all_request(
+        self,
+        resource_admin_extract_mock: Mock,
+    ):
+        resource_admin_extract_mock.return_value = (
+            TestResourceConstants.TEST_EXTRACT_RESOURCE_GET_AUDIT_RULES
+        )
+        self.assertEqual(
+            self.resource_admin.alter_audit_by_attempt(
+                "TESTING", "ELIJTEST", failure="control", success="alter", all="read"
+            ),
+            TestResourceConstants.TEST_RESOURCE_ALTER_AUDIT_BY_ATTEMPT_ALL_REQUEST_XML,
+        )
+
+    def test_resource_admin_build_alter_audit_by_attempt_value_error(self):
+        bad_success = "problem"
+        bad_failure = "improper"
+        bad_all = "bad"
```

Comment:
It might be good to set one of these values to an integer or something to make sure your checks can handle that. For example, running the `.lower()` on an integer will result in an `AttributeError` and you want to make sure your code can handle that.

Reply:
new code handles that and I changed the test to take in an int for `bad_all` to verify that is handled as expected.